extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -1554,13 +1552,24 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
 
         try {
           // eslint-disable-next-line no-await-in-loop
-          await server.submitTransaction(builtMergeTransaction);
+          await submitTx({ server, tx: builtMergeTransaction });
         } catch (e) {
           console.error(e);
           migratedAccount.isMigrated = false;
         }
       }
 
+      if (migratedAccount.isMigrated) {
+        // replace the source account with the new one in `allAccounts` and store the keys
+        // eslint-disable-next-line no-await-in-loop
+        await _replaceAccount({
+          mnemonicPhrase: migratedMnemonicPhrase,
+          password,
+          keyPair: newKeyPair,
+          indexToReplace: keyIdIndex,
+        });
+      }
```

Comment:
don't run this method until we've confirmed that the migration was completely successful